### PR TITLE
Update minitwit & improve testing for examples

### PR DESCRIPTION
**Pull request by wgwz**
* Related to #1945
* Re-works minitwit to be installed and run as:

    pip install --editable .
    export FLASK_APP=minitwit.minitwit
    export FLASK_DEBUG=1
    flask initdb
    flask run

* added flaskr and minitwit to norecursedirs
  * tests not properly run when using pytest standards
  * see: http://stackoverflow.com/questions/38313171/configuring-pytest-with-installable-examples-in-a-project
* Both flaskr and minitwit now follow pytest standards.
* Tests can for them as `py.test` or `python setup.py test`

### CODE REVIEW

**Comment by wgwz** [comment]
@davidism heyo. how does this stuff look? pretty minimal changes here

File: examples/flaskr/setup.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 setup(
     name='flaskr',
-    packages=['flaskr'],
+    packages=['flaskr',],
```
Comment (untitaker) [inline examples/flaskr/setup.py]: This is not needed.

File: docs/tutorial/dbcon.rst [hunk 2]
```diff
@@ -8,7 +8,7 @@ You now have a function for establishing a database connection with
 closing database connections all the time is very inefficient, so you will
 need to keep it around for longer.  Because database connections
 encapsulate a transaction, you will need to make sure that only one
-request at a time uses the connection. An elegant way to do this is by
```
Comment (untitaker) [inline docs/tutorial/dbcon.rst]: Those are unrelated changes, could you please file a separate PR? Thanks!

File: examples/minitwit/setup.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+from setuptools import setup
+
+setup(
+	name='minitwit',
```
Comment (ThiefMaster) [inline examples/minitwit/setup.py]: please du not use tabs for indentation.

**Comment by untitaker** [comment]
I think the `minitwit.py` should be `__init__.py`.

**Comment by wgwz** [comment]
@untitaker That seems like it might be confusing for a new user. Is that a common pattern (moving the app into __init__.py)?

File: examples/minitwit/setup.py (hunk 3 above)
Comment (wgwz) [inline examples/minitwit/setup.py]: my bad. i should have been more careful with that. lesson learned :)

File: docs/tutorial/dbcon.rst (hunk 2 above)
Comment (wgwz) [inline docs/tutorial/dbcon.rst]: @untitaker will do.

Comment (wgwz) [inline docs/tutorial/dbcon.rst]: @untitaker I'd like you to take a second look at this. I think the changes are related enough that there is not really a need to file a separate PR. `flaskr` and `minitwit` will be/are structured exactly the same way. Therefore the documentation is really relevant to both examples. Most of the changes are about testing, so it's not just superficial grammar stuff. (I'm not being lazy, I just don't agree that they are unrelated :P)

**Comment by wgwz** [comment]
@untitaker Adding `from minitwit import app` into the *\_\_init\_\_.py* has the effect of `$ export FLASK_APP=minitwit` working. Which I agree looks much cleaner than `$ export FLASK_APP=minitwit.minitwit`. What do you think about that change?

**Comment by untitaker** [comment]
That's exactly the effect I'd like to have. Why do you think that having code in `__init__.py` is confusing for newbies?

**Comment by wgwz** [comment]
I don't mind having the one-line import statement in the *\_\_init\_\_.py*. I just think that having all of *minitwit.py* inside of the *\_\_init\_\_.py* will obfuscate the *\_\_init\_\_* files purpose for a newcomer.

**Comment by wgwz** [comment]
@untitaker What do you think about [this](https://github.com/pallets/flask/pull/1954#discussion_r73083362)? (Not trying to be a nuisance, I'm just thinking it might have got lost in your notifications.)

**Comment by untitaker** [comment]
Does somebody else have opinions on this topic? @davidism @ronnypfannschmidt

**Comment by wgwz** [comment]
Hi, all. Any more comments on this? I'll have some time this week to work on things. If anyone can offer some guidance, I'd be glad to go ahead with fixing this PR.

**Comment by untitaker** [comment]
Im on mobile right now, but there are no more comments from my side I think.

The comments that are not outdated still apply. I don't really agree about the change being related, and I'd rather apprechiate it if it wouldn't be included, not because it is controversial in any way, but because it makes proper review much harder.

**Comment by wgwz** [comment]
@untitaker No problem, I'll go ahead making your suggested changes then. Thanks for the feedback.

**Comment by untitaker** [comment]
@wgwz Any update on this?

**Comment by wgwz** [comment]
@untitaker I'm not sure if I went about reverting the files in the best way possible. My branch has was based off the 0.11-maintenance branch. So I just ran `git checkout <commit ref prior to my first commit> -- docs/tutorial/<file-to-revert>.rst` for each of the files being reverted. I double-checked and the changes seem to be reverted correctly.

**Comment by untitaker** [comment]
@wgwz You asked on IRC: There is one comment left on the ["Files changed" tab](https://github.com/pallets/flask/pull/1954/files).

**Comment by wgwz** [comment]
@untitaker Sorry I missed that one somehow but it is taken care of now. How does it look now?

**Comment by untitaker** [comment]
A+ would merge again

**Comment by wgwz** [comment]
@untitaker Thanks :) I have to send another PR for flaskr. In order to match testing with how it's done in this PR for minitwit. And to add the import line in the init file. Also I will send a separate PR with the tutorial doc changes I removed from this one.

**Comment by untitaker** [comment]
@wgwz Thanks again, and feel free to directly mention me on those.
